### DISCUSSION TITLE
Financials or fundamental data

### DISCUSSION THREAD

**Discussion by ArunTS96** [BennyThadikaran/eod2#103] [General]
Is it possible to get last four quarter sales, eps, etc?
![image](https://github.com/BennyThadikaran/eod2/assets/10785357/3edbab40-4834-43c0-a180-ebe8f4119be3)

Attaching image from NSE.

**Comment by BennyThadikaran** [comment]
Hi Arun,

In Firefox, use `CTRL +  SHIFT + e` to open Network Inspector, click `reload` and filter all ajax requests by clicking on `XHR`
![filter_xhr](https://github.com/BennyThadikaran/eod2/assets/19973418/8421b856-c8ec-4c5f-a52b-2f9a3939e8ef)

Keep the inspector open and navigate to the Financial Results page you mentioned. You should find the below request which is the result you're looking for.
![nse-get](https://github.com/BennyThadikaran/eod2/assets/19973418/11b5aa76-ed9b-4c4e-874e-00d20bae91a0)

Click on it to get additional info. Right click on the request > Copy Value > Copy as Curl. You can paste this in your command line to make a curl request if you have it installed. Or go to https://curlconverter.com/ and paste the contents there. You should get the Python requests equivalent code.

Below is the concise json response. 

```json
[
  {
    "from_date": "01 Oct 2023",
    "to_date": "31 Dec 2023",
    "expenditure": "5807235",
    "income": "8171965",
    "audited": "Un-Audited",
    "cumulative": "Non-cumulative",
    "consolidated": "Non-Consolidated",
    "reDilEPS": "21.58",
    "reProLossBefTax": "1943066",
    "proLossAftTax": "1637254",
    "re_broadcast_timestamp": "16-Jan-2024 17:11",
    "xbrl_attachment": "https://nsearchives.nseindia.com/corporate/xbrl/BANKING_101078_1023175_16012024051131.xml",
    "na_attachment": "https://nsearchives.nseindia.com/na_attachments/HDFCBANK_NA_16012024171130_1.zip"
  }
]
```

**Reply by ArunTS96** [reply]
Awesome Benny. Will try it out.
